WebService/v1/Controllers/ValuesController.cs

```diff
@@ -95,7 +95,7 @@ private void EnsureValidId(string collectionId, string key = "")
         {
             // Currently, there is no official document describing valid character set of document ID
             // We just verified and enabled characters below
-            var validCharacters = "_-";
+            var validCharacters = "_-.";
```

Comment:
this new char might be problematic in URLs when using ASP.NET, historically the "." is used in filenames to separate the file extension and MS services tend to look for a file when the "." is present. I would avoid adding it